### PR TITLE
feat: Mean finance add yield bearing tokens

### PR DESCRIPTION
**Pull request by FiboApe**
## Description
Mean Finance got upgraded! 🔥 
We have reduced the amount of extra data that was on the subgraph, that is being handled here 29d6431a8f03aa7c8fee4d36c3e22718c713b3d1

Also adding tracking for multiple Mean Finance contract Versions. Currently it was always fetching balances for users from the lastest version. But on the Mean Finance UI we do allow to see/modify positions created in earlier versions of Mean Finance (The version mostly refer to contract deployments). This is adding support to the multiple subgraphs we have for each version too.

As the last commit and the longest: Mean Finance now supports creating DCA Positions that generate yield while they are getting swapped/ wait for you to withdraw, what this means  ( 🌝 ) is that now instead of doing DCA from USDC -> ETH you could be doing it to aUSDC -> aWETH (Aave deposited USDC and ETH), so now you gain yield while the funds are deposited on mean finance. Since the aave token is a rebased token we had to create a wrapper for it to conform to ERC-4626, thats the ERC4626WRAP_ADDRESSES that you are going to find on the code. To show the balances we need to convert those wrapped ERC4626 into their underlying so as to show them on zapper as generating yield on AAVE

## Checklist

- [X] I have followed the [Contributing Guidelines](../CONTRIBUTING.md)
- [x] (optional) As a contributor, my Ethereum address/ENS is: fiboape.eth
- [x] (optional) As a contributor, my Twitter handle is: @fiboape

## How to test?
http://localhost:5001/apps/mean-finance/balances?addresses[]=0xf488aaf75D987cC30a84A2c3b6dA72bd17A0a555&network=polygon

### CODE REVIEW

File: src/apps/mean-finance/helpers/addresses.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    [Network.OPTIMISM_MAINNET]: 'https://api.thegraph.com/subgraphs/name/mean-finance/dca-v2-ys-optimism',
+    [Network.POLYGON_MAINNET]: 'https://api.thegraph.com/subgraphs/name/mean-finance/dca-v2-ys-polygon',
+  },
+  [POSITION_VERSION_4]: {
+    [Network.POLYGON_MAINNET]: 'https://api.thegraph.com/subgraphs/name/mean-finance/dca-v2-yf-polygon',
+    [Network.OPTIMISM_MAINNET]: 'https://api.thegraph.com/subgraphs/name/mean-finance/dca-v2-yf-optimism',
+  },
+};
+
+// Mean Finance Wrapped-4626 -> Unwrapped 4626
```
Comment (0xsambugs) [inline src/apps/mean-finance/helpers/addresses.ts:72]:
```suggestion
// Mean Finance Wrapped-4626 -> Unwrapped yield-bearing token
```

File: src/apps/mean-finance/optimism/mean-finance.balance-fetcher.ts [hunk 2]
```diff
@@ -21,59 +23,107 @@ const network = Network.OPTIMISM_MAINNET;
...
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(MeanFinanceContractFactory)
+    private readonly meanFinanceContractFactory: MeanFinanceContractFactory
+  ) { }
 
-  async getUserPositions(address: string) {
+  async getUserPositions(address: string, version: PositionVersions) {
     const graphHelper = this.appToolkit.helpers.theGraphHelper;
-    const data = await getUserPositions(address.toLocaleLowerCase(), Network.OPTIMISM_MAINNET, graphHelper);
+    const data = await getUserPositions(address.toLocaleLowerCase(), Network.OPTIMISM_MAINNET, graphHelper, version);
```
Comment (0xsambugs) [inline src/apps/mean-finance/optimism/mean-finance.balance-fetcher.ts:34]:
```suggestion
    const data = await getUserPositions(address.toLowerCase(), Network.OPTIMISM_MAINNET, graphHelper, version);
```

**Review by 0xsambugs — APPROVED** [review]
Looks good @FiboApe 👍

**Review by immasandwich — APPROVED** [review]
Awesome. Excited to try out the updated protocol too.

**Pull request merged**